### ISSUE TITLE
WebUI - Admin management of user SSH keys

### ISSUE THREAD

**Issue by yaleman** [kanidm/kanidm#1689]
**Is your feature request related to a problem? Please describe.**

Admins should be able to do the following with user SSH keys:

- [ ] list
- [ ] add
- [ ] delete

Based on the permissions available, of course. Similar to #1687

**Comment by Firstyear** [comment]
I'm a bit on the side of this being cli only for admins.

**Comment by yaleman** [comment]
This is intended for admin (and analogues) to view/manage *all* keys they have access to, either their own or other account types.

Limiting it to CLI is entirely arbitrary considering it's the same API at the end of the day, and the Web UI removes the need for folks to have the `kanidm` CLI tool, which while it's relatively easy to install, isn't always going to be available to users.

**Comment by Firstyear** [comment]
Yes it is arbitrary, but the arbitrary line is dev time - if I start to add all the admin functions to the webui and the cli that will take extra time, and we'll have to add all the existing functions, and it also makes it more work to change them. For now I've done it this way simply to save development time, and because most admins at least from the unix side will have a cli available - in fact it's impossible to start with Kani as an admin *without* a cli. It's kind of the same issue that happened in the past with the tui that was proposed, was it added surface area to development. That's why I've been hesitant to add admin functions to the UI - yet. 

I think also if we were to add these functions, we should consider a rearchitect of the webui into parts to help with latency reduction too.

**Comment by yaleman** [comment]
Oh, for sure - it's definitely not something I'd expect *you* to do specifically, it's something on the list that can be done by anyone, over time. 😄 

And yes, the huge-WASM-blob thing is a problem that needs investigating, and I still can't find a way of splitting a yew project into "modules" that actually saves significant bytes, nor can I figure out a way of making the "needed" functionality simple to develop against... yet.

**Comment by ToxicMushroom** [comment]
Since ssh keys will become part of cred-update flow, admins will need to start the cred update flow mimicking another user. I think this is possible already, but let me know if not :p. And otherwise I'll find out soon

**Comment by yaleman** [comment]
I would not want to encourage admins to be doing cred update flows on behalf of users -  that's the point of the existing reset links. 

Obviously they can already issue these links and run them themselves, but for troubleshooting/helpdesk purposes the "list ssh keys" should be visible to admins as an option, and removing an authentication method for security response reasons should be possible in the case of a compromised key.
